### ISSUE TITLE
pycodestyle incorrectly warns about missing whitespace around operators

### ISSUE THREAD

**Issue by thanatos** [PyCQA/pycodestyle#658]
Given the following `example.py`,

```python
my_list = [1, 2, 3]
a = 2+2
b = my_list[1:1+1]
```

`pycodestyle` incorrectly warns about the missing whitespace for the `+` operator inside the list slice:

```
» pycodestyle --select E226 ./example.py
./example.py:2:6: E226 missing whitespace around arithmetic operator  # OK
./example.py:3:16: E226 missing whitespace around arithmetic operator  # incorrect
```

The first warning (line 2) is correct (line 2 should read `a = 2 + 2`); the second warning (line 3) is incorrect according to PEP-8:

> Yes:
>
> ```python
> ham[1:9], ham[1:9:3], ham[:9:3], ham[1::3], ham[1:9:]
> ham[lower:upper], ham[lower:upper:], ham[lower::step]
> ham[lower+offset : upper+offset]
> ham[: upper_fn(x) : step_fn(x)], ham[:: step_fn(x)]
> ham[lower + offset : upper + offset]
> ```
>
> No:
>
> ```python
> ham[lower + offset:upper + offset]
> ham[1: 9], ham[1 :9], ham[1:9 :3]
> ham[lower : : upper]
> ham[ : upper]
> ```

**Comment by sigmavirus24** [comment]
What `pycodestyle` wants (and what I believe is in the `yes` section you listed is this:

```
b = my_list[1:1 + 1]
```

While `1+1` is allowed in PEP-0008, pycodestyle doesn't have the greatest ability to determine when the arithmetic in question is part of a an indexing operation or otherwise. It would require a state machine which it doesn't really have.

**Comment by Kache** [comment]
I was going to report the false negatives for `E225` and `E226`:

```py
# detected
a = 2-1
b = 2<1

# not detected
c = f"{2-1}"
d = f"{2<1}"
```

But perhaps this is also a "not a state machine" problem?

**Comment by asottile** [comment]
I believe those will be reported in 3.12+ -- they're invisible otherwise
